### DISCUSSION TITLE
Issue with a dictionary and a get function

### DISCUSSION THREAD

**Discussion by Sanchoyzer** [python/typing#1473] [Q&A]
```python
def test_me(my_dict: dict | None = None) -> None:
    def _f(func: Callable) -> None:
        func('a')

    if my_dict is not None:
        _f(lambda x: my_dict.get(x))
```
There is an error: `error: Item "None" of "dict[Any, Any] | None" has no attribute "get"  [union-attr]`. It looks a little bit strange because I've created a check for the 'None' case. What should I do to fix it?

P.S. `mypy==1.5.1`, `Python 3.11.4`

**Comment by gandhis1** [comment]
This is an interesting example. There was a change a few months ago that propagated narrowed types to nested functions: https://github.com/python/mypy/pull/15133 (fixing https://github.com/python/mypy/issues/2608)

Perhaps lambdas were not covered there?

**Reply by erictraut** [reply]
FWIW, this type checks without errors in pyright.

I recommend filing a bug report in the [mypy issue tracker](https://github.com/python/mypy/issues).

**Reply by Sanchoyzer** [reply]
Ok, I've created the [mypy issue](https://github.com/python/mypy/issues/16148)
